### PR TITLE
 Added 'token_details' for context()

### PR DESCRIPTION
**Pull request by nipampatel**

### CODE REVIEW

**Comment by nipampatel** [comment]
@jonathankeebler @jpdemiranda @rcjpisani Please review.

File: lib/index.js [hunk 1]
```diff
@@ -286,6 +286,16 @@ class Provider extends EventEmitter
 							resolve(res);
 						});
 				});
+            },
+            token_details:() => {
+				return new Promise((resolve, reject) => {
+					this.auth.get_token_details(token).then(function(token_details) {
+						resolve({
+                            user_id : token_details.user_id,
```
Comment (rcjpisani) [inline lib/index.js]: Can we fix the indentation (use tabs) for lines 289-295?

File: lib/index.js [hunk 2]
```diff
@@ -312,8 +322,8 @@ class Provider extends EventEmitter
 
 	on(event_name, permissions, next)
 	{
-		const provider = this;
-
+        const provider = this;
```
Comment (rcjpisani) [inline lib/index.js]: Can we fix the indentation here as well (use tabs)

File: lib/index.js [hunk 3]
```diff
@@ -312,8 +322,8 @@ class Provider extends EventEmitter
...
+
```
Comment (rcjpisani) [inline lib/index.js]: Remove white space here

File: lib/index.js (hunk 1 above)
Comment (nipampatel) [reply lib/index.js]: @rcjpisani I tried :). Can you please have a look again?

File: lib/index.js (hunk 2 above)
Comment (nipampatel) [reply lib/index.js]: Removed. Please review again.

**Pull request merged**